### PR TITLE
add /LOAD/PCYL compatible with /H3D/ELEM/PEXT

### PR DESCRIPTION
**Pull request by qiangRadioss**
#### Description of the changes

make compatible output /H3D/ELEM/PEXT for new /LOAD/PCYL

### CODE REVIEW

File: engine/source/input/lectur.F [hunk 1]
```diff
@@ -192,6 +193,7 @@ SUBROUTINE LECTUR(ICODE   ,ISKEW ,ISKWN  ,IXTG   ,IXS   ,
       TYPE (SENSORS_) ,INTENT(INOUT) ,TARGET :: SENSORS
       TYPE (DYNAIN_DATABASE), INTENT(INOUT)  :: DYNAIN_DATA
       TYPE (DT_), INTENT(INOUT)              :: DT
+      TYPE (LOADS_)                          :: LOADS
```
Comment (sebastienVilleneuve) [inline engine/source/input/lectur.F:196]: Hi @qiangRadioss 

INTENT statement is missing here

File: engine/source/output/h3d/h3d_results/genh3d.F [hunk 2]
```diff
@@ -249,7 +252,7 @@ SUBROUTINE GENH3D(X        ,D        ,V           ,A           ,BUFEL    ,
      .        NERBE3T(NRBE3G),IAD_RBE2(4,*),NV46,KXIG3D(*),
      .        IXIG3D(*),SIG3DSOLID,MDS_MATID(*)
       INTEGER LLOADP(SLLOADP)
-      INTEGER ILOADP(SIZLOADP,*),IBCL(NIBCLD,*)
+      INTEGER ILOADP(SIZLOADP,*),IBCL(NIBCLD,*),IFRAME(LISKN,*)
```
Comment (sebastienVilleneuve) [inline engine/source/output/h3d/h3d_results/genh3d.F:255]: INTENT  statement is missing

Comment (sebastienVilleneuve) [inline engine/source/output/h3d/h3d_results/genh3d.F:255]: "*" is not allowed in dimensions

File: engine/source/output/h3d/h3d_results/h3d_skin_ixskin.F [hunk 3]
```diff
@@ -795,6 +860,30 @@ SUBROUTINE H3D_PRE_SKIN_IXSKIN(NODAL_IPART,IB ,ILOADP,LLOADP,
...
+C---------/LOAD/PCYL	   
+       NSKIN1 = NSKIN
+       DO NL=1,LOADS%NLOAD_CYL
+         DO I = 1,LOADS%LOAD_CYL(NL)%NSEG
+           N1 = LOADS%LOAD_CYL(NL)%SEGNOD(I,1)
+           N2 = LOADS%LOAD_CYL(NL)%SEGNOD(I,2)
+           N3 = LOADS%LOAD_CYL(NL)%SEGNOD(I,3)
+           N4 = LOADS%LOAD_CYL(NL)%SEGNOD(I,4)
+C          
+		   NP = NP + 1
```
Comment (sebastienVilleneuve) [inline engine/source/output/h3d/h3d_results/h3d_skin_ixskin.F:872]: indentation to correct

File: engine/source/output/h3d/h3d_results/h3d_skin_scalar.F [hunk 4]
```diff
@@ -92,6 +94,9 @@ SUBROUTINE H3D_SKIN_SCALAR(
       CHARACTER*ncharline KEYWORD
       TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) ,INTENT(IN) :: SENSOR_TAB
       TYPE (H3D_DATABASE) :: H3D_DATA
+      TYPE (LOADS_)   , INTENT(IN) :: LOADS
+      INTEGER ,       DIMENSION(LISKN,*)  ,INTENT(IN) :: IFRAME
```
Comment (sebastienVilleneuve) [inline engine/source/output/h3d/h3d_results/h3d_skin_scalar.F:98]: "*" not allowed

File: engine/source/output/h3d/h3d_results/h3d_skin_scalar.F [hunk 5]
```diff
@@ -563,10 +570,13 @@ SUBROUTINE H3D_PRE_SKIN_SCALAR(SKIN_SCALAR,NODAL_IPART,
       my_real
      .   FAC(LFACCLD,*),XFRAME(NXFRAME,*),FORC(LFACCLD,*)
       TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) ,INTENT(IN) :: SENSOR_TAB
+      TYPE (LOADS_)   , INTENT(IN) :: LOADS
+      INTEGER ,       DIMENSION(LISKN,*)  ,INTENT(IN) :: IFRAME
```
Comment (sebastienVilleneuve) [inline engine/source/output/h3d/h3d_results/h3d_skin_scalar.F:574]: "*" not allowed

File: engine/source/output/h3d/h3d_results/h3d_skin_scalar.F [hunk 6]
```diff
@@ -914,24 +920,112 @@ SUBROUTINE H3D_PRE_SKIN_SCALAR(SKIN_SCALAR,NODAL_IPART,
...
+         ISIZ_SEG  = LOADS%LOAD_CYL(NL)%NSEG 
+         DO N=1, ISIZ_SEG
+           N1 = LOADS%LOAD_CYL(NL)%SEGNOD(N,1)
+           N2 = LOADS%LOAD_CYL(NL)%SEGNOD(N,2)
+           N3 = LOADS%LOAD_CYL(NL)%SEGNOD(N,3)
+           N4 = LOADS%LOAD_CYL(NL)%SEGNOD(N,4)
+		   IF (N1==0.OR.N2==0.OR.N3==0) CYCLE
+           NSKIN = NSKIN0+ IMAPSKP(NP0+N)
+           IF (NODAL_IPART(N1)>0) THEN
+		     IF (H3D_PART(NODAL_IPART(N1))==1) IS_WRITTEN_SKIN(NSKIN)=1
```
Comment (sebastienVilleneuve) [inline engine/source/output/h3d/h3d_results/h3d_skin_scalar.F:943]: some tabulations to remove

File: engine/source/output/h3d/h3d_results/h3d_skin_scalar.F [hunk 7]
```diff
@@ -914,24 +920,112 @@ SUBROUTINE H3D_PRE_SKIN_SCALAR(SKIN_SCALAR,NODAL_IPART,
...
+C---------/LOAD/PCYL	
+       DO NL=1,LOADS%NLOAD_CYL
+C--------default zero        
+         ISIZ_SEG  = LOADS%LOAD_CYL(NL)%NSEG 
+         DO N=1, ISIZ_SEG
+           N1 = LOADS%LOAD_CYL(NL)%SEGNOD(N,1)
+           N2 = LOADS%LOAD_CYL(NL)%SEGNOD(N,2)
+           N3 = LOADS%LOAD_CYL(NL)%SEGNOD(N,3)
+           N4 = LOADS%LOAD_CYL(NL)%SEGNOD(N,4)
+		   IF (N1==0.OR.N2==0.OR.N3==0) CYCLE
```
Comment (sebastienVilleneuve) [inline engine/source/output/h3d/h3d_results/h3d_skin_scalar.F:940]: tabulations to remove

File: engine/source/output/h3d/h3d_results/h3d_skin_vector.F [hunk 8]
```diff
@@ -83,10 +85,13 @@ SUBROUTINE H3D_SKIN_VECTOR(SKIN_VECTOR,NODAL_IPART,NSENSOR,
       my_real
      .   FAC(LFACCLD,*),XFRAME(NXFRAME,*),FORC(LFACCLD,*)
       TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) ,INTENT(IN) :: SENSOR_TAB
+      TYPE (LOADS_)   , INTENT(IN) :: LOADS
+      INTEGER ,       DIMENSION(LISKN,*)  ,INTENT(IN) :: IFRAME
```
Comment (sebastienVilleneuve) [inline engine/source/output/h3d/h3d_results/h3d_skin_vector.F:89]: "*" not allowed

File: engine/source/output/h3d/h3d_results/h3d_skin_vector.F [hunk 9]
```diff
@@ -438,16 +446,113 @@ SUBROUTINE H3D_SKIN_VECTOR(SKIN_VECTOR,NODAL_IPART,NSENSOR,
...
+         ISIZ_SEG  = LOADS%LOAD_CYL(NL)%NSEG 
+         DO N=1, ISIZ_SEG
+           N1 = LOADS%LOAD_CYL(NL)%SEGNOD(N,1)
+           N2 = LOADS%LOAD_CYL(NL)%SEGNOD(N,2)
+           N3 = LOADS%LOAD_CYL(NL)%SEGNOD(N,3)
+           N4 = LOADS%LOAD_CYL(NL)%SEGNOD(N,4)
+           IF (N1==0.OR.N2==0.OR.N3==0) CYCLE
+           NSKIN = NSKIN0+ IMAPSKP(NP0+N)
+           IF (NODAL_IPART(N1)>0) THEN
+		     IF (H3D_PART(NODAL_IPART(N1))==1) IS_WRITTEN_SKIN(NSKIN)=1
```
Comment (sebastienVilleneuve) [inline engine/source/output/h3d/h3d_results/h3d_skin_vector.F:470]: tabulation to remove

**Review by sebastienVilleneuve — CHANGES_REQUESTED** [review]

**Comment by qiangRadioss** [comment]
done after the comments.
thanks.

**Review by sebastienVilleneuve — APPROVED** [review]

**Pull request merged**